src/components/LeftNav.js

```diff
@@ -34,64 +46,170 @@ const TitleText = styled.Text`
+    } else {
+      this.count++
+      this.timeout = setTimeout(() => {
+        if (this.count === 2) {
+          this.setState({
+            edit: true,
+          })
+        }
+        this.count = 0
+      }, 250)
+    }
+  }
+
+  getFromArchive = (key: string) => {
+    const note = this.props.archive.find(note => note.key === key)
+    return note
+  }
+
+  updateText = text => {
+    this.setState({ newTitle: text })
+  }
+
+  updateFolder = text => {
+    this.setState({ newFolder: text })
+  }
+
+  onDragOver = e => {
+    e.preventDefault()
+  }
+
+  onDragStart = (e, key) => {
+    e.dataTransfer.setData('key', key)
+  }
+
+  onDrop = (e, folder) => {
+    const key = e.dataTransfer.getData('key')
+    const note = Object.assign(
+      {},
+      this.props.getNote(key)
+        ? this.props.getNote(key)
+        : this.getFromArchive(key),
+    )
+    !this.props.getNote(key) && this.props.updateArchive(note)
+
+    note.folder = folder
+    this.props.update(note, true)
+  }
+
+  archiveNote = e => {
+    const key = e.dataTransfer.getData('key')
+    const note = Object.assign({}, this.props.getNote(key))
+    this.props.updateArchive(note)
+  }
+
   render() {
     return (
       <Container>
         <SearchContainer>
-          <SearchBar data={this.props.notes || null} />
+          <SearchBar data={this.props.notes} />
+          <NewButton
+            onPress={() =>
+              this.props.update({
+                key: uuidv4(),
+                date: new Date().getTime(),
+              })
+            }
+            title="Add new note"
+          />
         </SearchContainer>
-        <TitleText>Your Recent Notes</TitleText>
-        <FlatList
-          data={
-            this.props.notes.length < 5
-              ? this.props.notes
-                  .sort((a, b) => {
-                    return b.date - a.date
-                  })
-                  .slice(0, this.props.notes.length)
-              : this.props.notes
-                  .sort((a, b) => {
-                    return b.date - a.date
-                  })
-                  .slice(0, 5)
-          }
-          renderItem={({ item }) => (
-            <Text onClick={() => this.props.update(item)}>{item.title}</Text>
-          )}
-        />
         <TitleText>Your Notes</TitleText>
-        <FlatList
-          data={this.props.notes.reverse()}
-          renderItem={({ item }) => (
-            <Text onClick={() => this.props.update(item)}>{item.title}</Text>
-          )}
+        <NewButton title="Add a new folder" onPress={this.addFolder} />
+        {Object.values(this.props.getFolders()).map(
+          (subArray: Array<Note>, index: number) => {
+            return (
+              <View key={subArray[0].key}>
+                <Folder
+                  data={subArray}
+                  openFolder={this.openFolder}
+                  folderID={subArray[0].key}
+                  folderName={subArray[0].folder && subArray[0].folder}
```

Comment:
Something looks off here :)

Reply:
so this is actually to satisfy flow since folder is an optional prop, but obviously it's a given in this particular array that there will be a folder.. what do you think is best?